tests/unit/test_discovery_client.py

```diff
@@ -278,7 +279,7 @@ def test___key_file_exist_after_poll___start_discovery_service___discovery_servi
 
     mocker.patch(
         "ni_measurementlink_service.discovery._support._open_key_file",
-        side_effect=[OSError, OSError, OSError, temp_discovery_key_file_path],
+        side_effect=[OSError, OSError, OSError, TextIOBase()],
```

Comment:
The [open_key_file](https://github.com/ni/measurementlink-python/blob/a295c1a890acfb9a411a5c8f0d9090e6eec20fa9/ni_measurementlink_service/discovery/_support.py#L126) API's expected output is of type [TextIOWrapper](https://docs.python.org/3/library/os.html#os.fdopen) (which is TextIO as per type hint) but since they're abstract class, I used its base class (`TextIOBase`) under `io` to create an instance of it.

Reply:
I'm surprised that `TextIOBase` is not abstract because [PEP 3116](https://peps.python.org/pep-3116/#text-i-o) (which introduced it) describes it as an abstract base class.

Also, `TextIOBase` doesn't actually subclass `TextIO`, at least in typeshed: https://github.com/python/typeshed/blob/main/stdlib/io.pyi#L136

Python provides (at least) two concrete implementations of `TextIO`: `TextIOWrapper` and `StringIO`. These are not abstract.

Returning a mock `TextIO` would also be acceptable, but in this case the simplest solution is to return a `StringIO`.

Reply:
> I'm surprised that TextIOBase is not abstract because [PEP 3116](https://peps.python.org/pep-3116/#text-i-o) (which introduced it) describes it as an abstract base class.

Okay, Sorry. I wasn't aware of this. I went ahead with returning `StringIO`.